base/dcp_client_metadata.go

```diff
@@ -82,17 +82,16 @@ func NewDCPMetadataMem(numVbuckets uint16) *DCPMetadataMem {
 }
 
 // Rollback resets the metadata, preserving EndSeqNo
-func (m *DCPMetadataMem) Rollback(ctx context.Context, vbID uint16) {
-	endSeqNo := m.metadata[vbID].EndSeqNo
+func (m *DCPMetadataMem) Rollback(ctx context.Context, vbID uint16, vbUUID gocbcore.VbUUID, startSeqNo gocbcore.SeqNo) {
 	m.metadata[vbID] = DCPMetadata{
-		VbUUID:          0,
-		StartSeqNo:      0,
-		EndSeqNo:        endSeqNo,
-		SnapStartSeqNo:  0,
-		SnapEndSeqNo:    0,
+		VbUUID:          vbUUID,
```

Comment:
Is there a safety check required that startSeqNo is less than EndSeqNo?  Or is that valid and the stream is just closed immediately by the remote.